R/create_hub_schema.R

```diff
@@ -238,7 +239,9 @@ coerce_datatype <- function(types) {
 }
 
 test_iso_date <- function(x) {
-  class(try(as.Date(x), silent = TRUE)) == "Date"
+  to_date <- try(as.Date(x), silent = TRUE)
+  isFALSE(inherits(to_date, "try-error")) &&
+    isTRUE(all.equal(which(is.na(x)), which(is.na(to_date))))
```

Comment:
To @nickreich's comment, I think this would provide effective semantics. 

```suggestion
  if (inherits(to_date, "try-error")) {
    return(FALSE)
  }
  missing_chars <- which(is.na(x))
  missing_dates <- which(is.na(to_date))
  isTRUE(all.equal(missing_chars, missing_dates))
```